pybricks/media/pb_type_image.c

```diff
@@ -0,0 +1,353 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025 The Pybricks Authors
+
+#include "py/mpconfig.h"
+
+#if PYBRICKS_PY_MEDIA && PYBRICKS_PY_MEDIA_IMAGE
+
+#include "py/mphal.h"
+#include "py/obj.h"
+
+#include <pybricks/media.h>
+#include <pybricks/parameters.h>
+#include <pybricks/util_mp/pb_obj_helper.h>
+#include <pybricks/util_mp/pb_kwarg_helper.h>
+
+#include <pbio/image.h>
+#include <pbio/int_math.h>
+
+#include <pbdrv/display.h>
+
+extern const mp_obj_type_t pb_type_Image;
+
+// pybricks.media.Image class object
+typedef struct _pb_type_Image_obj_t {
+    mp_obj_base_t base;
+    // TODO: avoid using MicroPython heap for large allocation, use a separate
+    // memory allocator and use a python object to handle memory release.
+    uint8_t *buffer;  // m_malloc'd buffer, may be shared, NULL for display.
+    pbio_image_t image;
+    bool is_display;
+} pb_type_Image_obj_t;
+
+static int get_color(mp_obj_t obj) {
+    uint8_t max = pbdrv_display_get_max_value();
+    if (obj == mp_const_none) {
+        return max;
+    }
+    if (mp_obj_is_int(obj)) {
+        return mp_obj_get_int(obj);
+    }
+    const pbio_color_hsv_t *hsv = pb_type_Color_get_hsv(obj);
+    int32_t v = pbio_int_math_bind(hsv->v, 0, 100);
```

Comment:
This is fine for now, but if we want to convert true color images to grayscale, then it works better to take hue and saturation into account as well. But that is one of those bigger discussion questions. :smile:

Reply:
> should we add a light gray and dark gray in Color?

Not yet. Let's hold off on that until we have a look at the APIs again and reconsider what would be best for both NXT and EV3.

Reply:
> This was just to accept a Color parameter. By the way, should we add a light gray and dark gray in Color?

You can do `Color.WHITE * 0.75` and `Color.WHITE * 0.25`.